### DISCUSSION TITLE
Interesting discussion

### DISCUSSION THREAD

**Discussion by JHeLiu** [dotnet/maui#5872] [Ideas]
If one day implementing XAML syntax becomes Razor syntax, writing UI will be much easier
1. I can dynamically traverse data on the UI and generate components
2. Whether it can be more attractive to front-end engineers and easier to accept
I don't know if this idea works

Datas
`
List<string> Names= new List<string>(){ "A","B","C"};
`

Razor 
```
@foreach (var item in Names)
{
     <StackPanel   Height="20">
            <DockPanel>
                <Border>
                    <Label Content="@item"></Label>
                </Border>
            </DockPanel>
        </StackPanel>
}
```

```
XAML   UI

  <StackPanel x:Name="stack"  Height="20">
            <DockPanel>
                <Border>
                    <Label Content="1"></Label>
                </Border>
            </DockPanel>
        </StackPanel>


 // .cs ?? 
new .... StackPanel  add DockPanel ?
```

Isn't Razor easier to bind data to

**Comment by BinaryAssault** [comment]
In XAML, you'd probably do something like what is below. Binding is pretty simple with XAML (and even Razor). I suggest reading up on MVVM or even MVU if you're interested in XAML design. 

``````
<CollectionView ItemsSource="{Binding Names}">
    <CollectionView.ItemTemplate>
        <DataTemplate>
            <Border>
                <Label Text="{Binding Name}"/>
            </Border>
        </DataTemplate>
        </CollectionView.ItemTemplate>
</CollectionView>
``````

Then, preferably in your ViewModel (using CommunityToolkit.Mvvm.ComponentModel):
``````
// can be a plain List instead of an ObservableCollection if you don't plan to update it in real time. 
[ObservableProperty]
private ObservableCollection<NameModel> _names= new ObservableCollection<NameModel>();
``````

**Reply by JHeLiu** [reply]
good

**Reply by saint4eva** [reply]
For Comet C# MVU, the databinding is automatically done. In Comet, the View is aware of databinding
